helix-rest/src/main/java/org/apache/helix/rest/server/resources/metadatastore/MetadataStoreDirectoryAccessor.java

```diff
@@ -167,11 +169,15 @@ public Response deleteMetadataStoreRealm(@PathParam("realm") String realm) {
    */
   @GET
   @Path("/sharding-keys")
-  public Response getShardingKeys(@QueryParam("prefix") String prefix) {
+  public Response getShardingKeys(@QueryParam("prefix") String prefix,
+      @DefaultValue("false") @QueryParam("groupByRealm") boolean groupByRealm) {
     try {
-      if (prefix == null) {
+      if (prefix == null && !groupByRealm) {
         // For endpoint: "/sharding-keys" to get all sharding keys in a namespace.
         return getAllShardingKeys();
+      } else if (prefix == null) {
+        // For request: "/sharding-keys?groupByRealm=true" to get sharding keys by realm.
+        return getAllShardingKeysByRealm();
```

Comment:
```
if (prefix == null) {
  if (groupByRealm) {
    return getAllShardingKeysByRealm();
  }
  return getAllShardingKeys();
}
```

would be a more concise and readable way of expressing this logic. what do you think?

Reply:
Maybe the ways we think of if-else condition branch are different. I've changed it to another way which may look easier to understand.